src/Servers/Kestrel/test/InMemory.FunctionalTests/Http3/Http3StreamTests.cs

```diff
@@ -2414,5 +2422,70 @@ public async Task UnsupportedControlStreamType(int typeId)
             await requestStream.ExpectHeadersAsync().DefaultTimeout();
             await requestStream.ExpectReceiveEndOfStream().DefaultTimeout();
         }
+
+        [Fact]
+        public async Task PostRequest_ServerReadsPartialAndFinishes_SendsBodyWithEndStream()
+        {
+            var startingTcs = new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
+            var appTcs = new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
+            var clientTcs = new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
+            var headers = new[]
+            {
+                new KeyValuePair<string, string>(HeaderNames.Method, "GET"),
+                new KeyValuePair<string, string>(HeaderNames.Path, "/"),
+                new KeyValuePair<string, string>(HeaderNames.Scheme, "http"),
```

Comment:
Aren't all Http3 requests https?
```suggestion
                new KeyValuePair<string, string>(HeaderNames.Scheme, "https"),
```

Reply:
That's true, but as far as these tests are concerned we're doing HTTP/3 over an unencrypted in-memory transport.